crates/pallet-bridge-pot-currency-swap/src/genesis_verifier.rs

```diff
@@ -0,0 +1,62 @@
+//! An implementation that verifies bridge pot balance values at genesis.
+
+use frame_support::{
+    sp_runtime::{
+        traits::{CheckedAdd, Convert, Zero},
+        ArithmeticError,
+    },
+    sp_std::marker::PhantomData,
+    traits::{Currency, Get},
+};
+
+use super::{Config, GenesisVerifier, Pallet};
+
+/// The implementation that requires bridge pot balance values be balanced.
+pub struct Balanced<Pallet>(PhantomData<Pallet>);
+
+/// The bridge pot balances logic related error kinds.
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// The arithmetic operation error.
+    Arithmetic(ArithmeticError),
+}
+
+impl<T: Config<I>, I: 'static> Balanced<Pallet<T, I>> {
+    /// A function to calculate expected [`Config::PotTo`] balance based on the provided list of
+    /// all [`Config::AccountIdFrom`] balances except [`Config::PotFrom`] balance.
+    pub fn calculate_expected_to_bridge_balance(
```

Comment:
I'd like this logic tested with chain configurations with non-equal `Currency::minimum_balance` values. The rationale for this is here I see that `CurrencyTo::minimum_balance()` is added.

Just as heads up, I would like to see the exact numbers hardcoded at the expectations in the tests, rather than them being computed as well.

For instance, do `500 /* ed from */ + 123 /* ed to */` instead of `CurrencyTo::minimum_balance() + CurrencyFrom::minimum_balance()` at the expectations.

The rationale for this is it would be a more explicit test that way, that would ensure that not only the calculations are tested, but also the underlying assumptions are matched against the effective calculations, and will help us keep an eye on this logic in the context of the possible future changes. This rationale could as well be added at the tests...

---

Adding these tests could be done just for this fn as the units, or for the whole runtime - whichever you find more fitting.

Reply:
Changed the mocked runtime and test with different existential deposit values - https://github.com/humanode-network/humanode/pull/700/commits/ade7c56c6c6ae24d02a0211a04e8ff1186b833d5